### PR TITLE
fix(curriculum): implement a lesson about the box-sizing property

### LINKED ISSUE #49156: learn-the-css-box-model-by-building-a-rothko-painting challenge - missing information about different css box models
### Describe the Issue

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-the-css-box-model-by-building-a-rothko-painting/

Challenge called "Learn the **css box model** by..."
There are bunch of steps which describes some parts of the box model, a lot of steps about changing colors etc
But it looks like there is no step describing that there are different models available like content-box, border-box etc
And in the next project https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-css-flexbox-by-building-a-photo-gallery/step-7 there is a phrase:

> Normalize your box model by creating a * selector and setting the box-sizing property to border-box as the value.

But there was no information before that about border-box and what is it and why we are setting it to border-box here

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-the-css-box-model-by-building-a-rothko-painting/

### Steps to Reproduce

Just go through challenge

### Recommended fix or suggestions

It will great if "Learn the css box model" project will include information about different box-sizing values 

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by CBID2**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #49156 

I explained the `box-sizing` property in two separate steps. One focuses on the `content-box` value and its purpose while the other focuses on the `border-box` value and how it differs from the former. Through these changes, I anticipate that campers would gain a better understanding of the steps.

Edit by @naomi-lgbt: Because this adds new steps, I've synced the changes to the `chinese` and `espanol` directories to please the tests. Once merged, I'll manually run the upload and download flows to get every language updated properly.

### CODE REVIEW

**Comment by CBID2** [comment]
Note: Apparently Step 6 is failing, but I never touched that step.

**Comment by shootermv** [comment]
i tried to figure out what is the reason for failing tests -
my guess (only guess) is - step 7 has a tests (hints) but - there is nothing that user required to do (editable)

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md
Comment (shootermv) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:29]: i think here is the problem that makes tests to fail -
each challenge file must have only one `# --seed--` region
and also only one `## --seed-contents`...
here you have it twice...
you must remove the duplicate ones

Comment (CBID2) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:24]: Done @shootermv

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md
Comment (CBID2) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md:74]: Done

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md
Comment (CBID2) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:53]: > here is some css stuff which need to be moved to `css` section

I'm confused

**Comment by moT01** [comment]
I believe the tests aren't running here because of the conflicts @shootermv. I would try fixing those and see if that works. The `meta.json` files were [changed recently to use objects instead of arrays](https://github.com/freeCodeCamp/freeCodeCamp/pull/50955). You will need to change the meta file to align with that. Let us know if you need any guidance or anything with it.

**Comment by CBID2** [comment]
> I believe the tests aren't running here because of the conflicts @CBID2. I would try fixing those and see if that works. The `meta.json` files were [changed recently to use objects instead of arrays](https://github.com/freeCodeCamp/freeCodeCamp/pull/50955). The tests stopped running on this PR after that PR got merged. You will need to change the meta file to align with that. Let us know if you need any guidance or anything with it.

Thanks @moT01! :) I'll let you know

File: curriculum/challenges/_meta/learn-css-flexbox-by-building-a-photo-gallery/meta.json
Comment (shootermv) [inline curriculum/challenges/_meta/learn-css-flexbox-by-building-a-photo-gallery/meta.json:110]: there is some mistakes in the json, here i pasting the corrected version:
`{
  "name": "Learn CSS Flexbox by Building a Photo Gallery",
  "isUpcomingChange": false,
  "usesMultifileEditor": true,
  "hasEditableBoundaries": true,
  "dashedName": "learn-css-flexbox-by-building-a-photo-gallery",
  "helpCategory": "HTML-CSS",
  "order": 6,
  "time": "5 hours",
  "template": "",
  "required": [],
  "superBlock": "2022/responsive-web-design",
  "challengeOrder": [
    {
      "id": "61537485c4f2a624f18d7794",
      "title": "Step 1"
    },
    {
     "id": "61537a8054753e2f12a1574",
     "title": "Step 2"
    },
    {
      "id": "61537bb9b1a29430ac15ad38",
     "title":"Step 3"
    },
    {
     "id":  "61537c5f81f0cf325b4a854c",
     "title":      "Step 4"
    },
    {
     "id": "61537c9eecea6a335db6da79",
     "title":"Step 5"
    },  
    {
     "id": "61537d86bdc3dd343688fceb",
     "title": "Step 6"
    },
    {
      "id": "6494da0daf5df5197963671d",
      "title": "Step 7"
    },
    {
     "id": "6493bc0d99879635209565aa",
     "title": "Step 8"
    },
    {
      "id": "615380dff67172357fcf0425",
      "title": "Step 9"
    },
    {
    "id": "615f171d05def3218035dc85",
   "title": "Step 10"
    },
    {
     "id": "6153893900438b4643590367",
     "title": "Step 11"
    },
    {
    "id": "6153897c27f6334716ee5abe",
     "title": "Step 12"
    },
    {
    "id": "615389bd81347947ea7ba896",
     "title": "Step 13"
    },
    {
     "id": "6153908a366afb4d57185c8d",
    "title": "Step 14"
    },
    {
     "id": "615392916d83fa4f02f7e2cf",
    "title": "Step 15"
    },
    {
     "id": "6153938dce8b294ff8f5a4e9",
     "title": "Step 16"
    },
    {
   "id": "6153947986535e5117e60615",
    "title": "Step 17"
    },
    {
    "id":  "61539e07e7430b528fbffe21",
     "title": "Step 18"
    },
    {
     "id": "61539f32a206bd53ec116465",
     "title": "Step 19"
    },
    {
     "id": "6153a04847abee57a3a406ac",
     "title": "Step 20"
    },
    {
     "id": "6153a3485f0b20591d26d2a1",
     "title": "Step 21"
    },
    {
     "id": "6153a3952facd25a83fe8083",
     "title": "Step 22"
    },
    {
     "id": "6153a3ebb4f7f05b8401b716",
     "title": "Step 23"
    }
  ]
}`

File: curriculum/challenges/_meta/learn-css-flexbox-by-building-a-photo-gallery/meta.json
Comment (shootermv) [inline curriculum/challenges/_meta/learn-css-flexbox-by-building-a-photo-gallery/meta.json:19]: there is one missing character there 'f'

**Comment by naomi-lgbt** [comment]
Needs #51181 in to unblock the CI.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md:29]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.gallery img')?.maxWidth, '350px');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615380dff67172357fcf0425.md:35]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.gallery img')?.height, '300px');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:27]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.textAlign, 'center');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:33]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.textTransform, 'uppercase');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:39]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.padding, '32px');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:45]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.backgroundColor, 'rgb(10, 10, 35)');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:51]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.color, 'rgb(255, 255, 255)');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:57]:
```suggestion
```
Don't need this log.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153893900438b4643590367.md:58]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.header')?.borderBottom, '4px solid rgb(253, 179, 71)');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153897c27f6334716ee5abe.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153897c27f6334716ee5abe.md:23]:
```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle('.gallery'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153897c27f6334716ee5abe.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153897c27f6334716ee5abe.md:29]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.gallery')?.display, 'flex');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:20]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('*')?.boxSizing, 'border-box');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6494da0daf5df5197963671d.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6494da0daf5df5197963671d.md:25]:
```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle('*')?.boxSizing);
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6494da0daf5df5197963671d.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6494da0daf5df5197963671d.md:31]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('*')?.boxSizing, 'content-box');
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
If we use specific assert methods, we'll get better error messages in the CI.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:71]: None of this CSS is present in the previous step. We should never automatically add code for campers, unless there is a specific reason and we are very explicit about it.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md
Comment (naomi-lgbt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:71]: If it is not necessary for the step, yes.

If we need this CSS to demonstrate the box model effectively, we should add steps where the camper adds it.

Comment (CBID2) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:71]: Oh I see. Can you help me remove it? I’m not near my computer at the moment and it’s hard to do it on the GH mobile app.

Comment (naomi-lgbt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md:71]: Sure. I'll dig into this.

**Comment by naomi-lgbt** [comment]
@CBID2 I've broken down the steps a bit more so that we aren't adding code to the seed.

I've also synced the changes to the espanol and chinese curricula, to hopefully please the tests. Feel free to take a look at my changes.

**Comment by CBID2** [comment]
> @CBID2 I've broken down the steps a bit more so that we aren't adding code to the seed.
> 
> I've also synced the changes to the espanol and chinese curricula, to hopefully please the tests. Feel free to take a look at my changes.

Hey @naomi-lgbt. There's a test called GitHub and it failed.

**Comment by naomi-lgbt** [comment]
Yes, that test failure is expected. That test checks that a PR does not modify translated files, but in this case it is necessary to do so.

**Comment by CBID2** [comment]
> Yes, that test failure is expected. That test checks that a PR does not modify translated files, but in this case it is necessary to do so.

Ahh, so what should be done to make it pass?

**Comment by naomi-lgbt** [comment]
Nothing. This test specifically only runs on PRs, and we will merge this PR with that test failing once it has the necessary approvals.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615389bd81347947ea7ba896.md
Comment (jdwilkin4) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615389bd81347947ea7ba896.md:3]: This is a duplicate of step 14.
I think you should delete this step since it is a duplicate.
You can use the [step tools](https://contribute.freecodecamp.org/#/how-to-work-on-practice-projects?id=delete-step) for that

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I went through the whole project.
It looks good. 👍 
Left a few comments

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615389bd81347947ea7ba896.md
Comment (CBID2) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/615389bd81347947ea7ba896.md:3]: @naomi-lgbt, what do you think?

File: curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json
Comment (jdwilkin4) [inline curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: It looks like you made changes to the wrong file here.

This is for a different project.

I would suggest reverting your last commit and then removing the duplicate step in the flexbox project

Comment (CBID2) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: Ohh ok

Comment (CBID2) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: > It looks like you made changes to the wrong file here.
> 
> This is for a different project.
> 
> I would suggest reverting your last commit and then removing the duplicate step in the flexbox project

Made the changes @jdwilkin4

Comment (jdwilkin4) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: The duplicate issue for the flexbox project has been resolved but you will still need to revert this file back to its original version because you have altered steps for another project. 

Basically, this PR should not make any changes to the meta.json file for the css box model project because this PR should only be focused on the flexbox project

Comment (CBID2) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: Ooh ok. Uhhh, which ones should revert again?

Comment (jdwilkin4) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: I think the easiest way would be to revert back to this commit here 587279bf4821ede456d54047d144eb910f12903e

Then you can make another commit to remove the duplicate from the correct project like you did in this last commit here 
https://github.com/freeCodeCamp/freeCodeCamp/pull/50795/commits/0f21fcbc566a233418a302923892c6a1d064a1b3

Comment (CBID2) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: So...revert commit 587279b?

Comment (jdwilkin4) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: yes, you should go back to that commit.
The duplicate will only temporarily appear again because you can always push up another commit to correct the issue in the correct project.

Remember, that this repo uses squash and merge so it doesn't matter how many commits you have in the PR

Comment (CBID2) [reply curriculum/challenges/_meta/learn-the-css-box-model-by-building-a-rothko-painting/meta.json:71]: Ah ok

**Review by jdwilkin4 — APPROVED** [review]
I have gone through the steps again and the new changes work now 👍

**Comment by CBID2** [comment]
> I have gone through the steps again and the new changes work now 👍 

Yay! Thanks @jdwilkin4

**Review by naomi-lgbt — APPROVED** [review]

**Comment by naomi-lgbt** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Comment by CBID2** [comment]
> Thank you for your contribution to the page! 👍
> 
> We are happy to accept these changes and look forward to future contributions. 🎉
> 
> 

This is awesome! :)

**Pull request merged**
